src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,524 @@
+            let root_id         = default();
+            let segments        = default();
+            let file            = Path{root_id,segments};
+            let name            = "Hardcoded".to_string();
+            let method_pointer  = MethodPointer{defined_on_type,file,name};
+            let definition      = method_pointer.into();
+            let call            = uuid::Uuid::new_v4();
+            Some(LocalCall{definition,call})
+        });
+        let result = self.push_breadcrumb(&local_call);
+
+        if is_new_breadcrumb {
+            result.as_ref().map(|(_, new_index)| self.get_breadcrumb(*new_index).map(|breadcrumb| {
+
+                // === User Interaction ===
+
+                let new_index  = *new_index;
+                let breadcrumb = breadcrumb.clone();
+                let network    = &breadcrumb.frp.network;
+                let frp_debug  = &self.frp_debug;
+                frp::extend! { network
+                    eval_ breadcrumb.frp.outputs.selected(
+                        frp_debug.select_breadcrumb.emit(new_index);
+                    );
+                }
+            }));
+        }
+        result
+    }
+
+    fn debug_pop_breadcrumb(&self) -> Option<(usize,usize)> {
+        debug!(self.logger,"Debug pop.");
+        self.pop_breadcrumb()
+    }
+
+    fn pop_breadcrumb(&self) -> Option<(usize,usize)> {
+        debug!(self.logger, "Popping {self.current_index.get()}");
+        if self.current_index.get() > 0 {
+            info!(self.logger, "Popping breadcrumb view.");
+            let old_index = self.current_index.get();
+            let new_index = old_index - 1;
+            self.current_index.set(new_index);
+            Some((old_index,new_index))
+        } else {
+            None
+        }
+    }
+
+    fn remove_breadcrumbs_history_beginning_from(&self, index:usize) {
+        for breadcrumb in self.breadcrumbs.borrow_mut().split_off(index) {
+            info!(self.logger, "Removing {breadcrumb.info.method_pointer.name}.");
+            breadcrumb.unset_parent();
+        }
+    }
+}
+
+impl display::Object for BreadcrumbsModel {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+
+
+// ===================
+// === Breadcrumbs ===
+// ===================
+
+/// The Breadcrumb's view used for visualizing the breadcrumbs and navigating them.
+#[derive(Debug,Clone,CloneRef,Shrinkwrap)]
+#[allow(missing_docs)]
+pub struct Breadcrumbs {
+    #[shrinkwrap(main_field)]
+    model   : Rc<BreadcrumbsModel>,
+    pub frp : Frp
+}
+
+impl Breadcrumbs {
+    /// Constructor.
+    pub fn new<'t,S:Into<&'t Scene>>(scene:S, focus_manager:&FocusManager) -> Self {
+        let frp     = Frp::new();
+        let model   = Rc::new(BreadcrumbsModel::new(scene,&frp,focus_manager));
+        let network = &frp.network;
+
+
+        // === Breadcrumb selection ===
+
+        frp::extend! { network
+
+            // === Selecting ===
```

Comment:
below some sections you use empty line, below some you do not. Please make it consistent (use it)